kmall.py

```diff
@@ -3507,8 +3542,16 @@ def _divide_rec(self, rec):
 
     def _pad_to_dense(self, arr, padval=999.0, maxlen=500, override_type=None, detectioninfo=False):
         """
-        Appends the minimal required amount of zeroes at the end of each
-        array in the jagged array `M`, such that `M` looses its jaggedness.
+        Appends the minimal required amount of zeroes at the end of each array in the jagged array `M`, such that `M`
```

Comment:
the padding isn't related to valid/invalid detections, its how I handle the ping record as a whole.  I take all the unique combinations of serial number/sector number/frequency, and divide the ping record by those indices.  So one MRZ datagram might end up being 3 ping records for me, see kmall._divide_rec.

As an example, for the file you provided (0007_20190513_154724_ASVBEN) I end up with the following ping records
(ping_systemid_sectornum_frequency)
ping_40_0_260000
ping_40_0_290000
ping_40_1_320000
ping_40_1_350000
ping_40_2_260000
ping_40_2_290000

each of these being stored independently.  It's admittedly weird, but it allows me to use the xarray Dataset object which requires the timestamps be unique, which in turn lets me to all the easy interpolation, etc.  I've found you have to isolate for serial number/sector number/frequency in order to avoid duplicate times in the record.  

Theoretically, the code I have will support non-sector based systems as well, it will just store as a single Dataset object.  If we move to include other file formats.

I used detection info later when I export to csv or to the sounding Dataset to flag soundings.